cloudsql/mysql/database-sql/cloudsql_test.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bytes"
+	"net/http/httptest"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
+)
+
+func TestIndex(t *testing.T) {
+	testutil.EndToEndTest(t)
+	tcpApp := createTCPApp()
+	loadIndex(tcpApp, t)
+	unixApp := createUnixApp()
+	loadIndex(unixApp, t)
+}
```

Comment:
The standard way to do this in Go is using a table driven test (https://github.com/golang/go/wiki/TableDrivenTests). See https://github.com/golang/go/wiki/TestComments#table-driven-tests-vs-multiple-test-functions for a longer form comparison to multiple test functions.

The goal is to make it easy to follow exactly what is being tested and how, including making it easy to debug.

In this case, it would look something like this:

```go
func testIndex(t *testing.T) {
	testutil.endToEndTest(t)
	tests := []struct{
		dbHost string
	}{
		{dbHost: ""},
		{dbHost: os.Getenv("DB_HOST")},
	}
	
	for _, test := range tests {
		oldDBHost := os.Getenv("DB_HOST")
		os.Setenv("DB_HOST", test.dbHost)
		
		a := newApp()
		rr := httptest.NewRecorder()
		// ...
		// Make sure to include the value of test.dbHost in any test failures.
		// Also see https://github.com/golang/go/wiki/TestComments#identify-the-function and Identify the Input right after.
		os.Setenv("DB_HOST", oldDBHost)
	}
}
```

Reply:
Done.